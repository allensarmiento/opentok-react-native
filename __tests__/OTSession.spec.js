import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Text } from 'react-native';

import OTSession from '../src/OTSession';

jest.mock('../src/OT', () => ({
  OT: {
    disconnectSession: jest.fn(),
    initSession: jest.fn(),
    connect: jest.fn()
  },
  setNativeEvents: jest.fn()
}));

describe('OTSession', () => {
  let apiKey, sessionId, token;

  beforeEach(() => {
    apiKey = 'fakeApiKey';
    sessionId = 'fakeSessionId';
    token = 'fakeToken';
  });

  describe('no props', () => {
    describe('missing credentials', () => {
      let component;
      console.error = jest.fn();
      console.log = jest.fn();

      beforeEach(() => {
        console.error.mockClear();
        component = mount(<OTSession />);
      });

      it('should render an empty view', () => {
        expect(toJson(component)).toMatchSnapshot();
      });

      it('should call console error', () => {
        expect(console.error).toHaveBeenCalled();
        expect(console.error).toHaveBeenCalledTimes(1);
      });
    });
  });

  describe('with props', () => {
    describe('with children', () => {
      it('should have two children', () => {
        const sessionComponent = renderer.create(
          <OTSession apiKey={apiKey} sessionId={sessionId} token={token}>
            <Text />
            <Text />
          </OTSession>
        ).toJSON();
        expect(sessionComponent).toMatchSnapshot();
      });
    });
  });
});