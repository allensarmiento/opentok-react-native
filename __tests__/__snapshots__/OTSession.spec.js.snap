// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OTSession no props missing credentials should render an empty view 1`] = `
<OTSession
  eventHandlers={Object {}}
  options={Object {}}
  signal={Object {}}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View />
  </View>
</OTSession>
`;

exports[`OTSession with props with children should have two children 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text />
  <Text />
</View>
`;
